test/e2e/util.go

```diff
@@ -0,0 +1,103 @@
+package e2e
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/console-operator/pkg/testframework"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+)
+
+// Basically each of these tests helpers are similar, they only vary in the
+// resource they are GETting, PATCHing and patch itself.
+// Since after the patch is done, the test needs to wait, till it can GET
+// patched object(or not, if the operator status is set to Managed).
+
+func patchAndCheckConfigMap(t *testing.T, client *testframework.Clientset) bool {
+	res, err := testframework.GetResource(client, "ConfigMap")
+	errorCheck(t, err)
+	configMap, ok := res.(*corev1.ConfigMap)
```

Comment:
The casting again concerns me a bit, I see it in all of these functions.  Wondering if its a signal the generalized `GetResource` ought to be a specific function for each resource type w/a specific return type?

Reply:
yeah was thinking about it and thought it would be better to have an single get function for all the types then multiple get functions for each type that would also cast object directly. Wasnt really sure which one is more idiomatic.